File: Lab1/Lab1/Lab1/model/Motherboard.cs
﻿using Lab1.model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab1Code.model
{
    class Motherboard : ElectronicСomponent
    {
        private MotherboardTypes motherboardTypes;
        private RAMTypes rAMType;
        private CPUTypes cPUTypes;
        private CPU cpu=null;
        private RAM ram=null;

        public MotherboardTypes MotherboardTypes { get => motherboardTypes; set => motherboardTypes = value; }
        public RAMTypes RAMType { get => rAMType; set => rAMType = value; }
        public CPUTypes CPUTypes { get => cPUTypes; set => cPUTypes = value; }
        internal CPU Cpu { get => cpu; set => cpu = value; }
        internal RAM Ram { get => ram; set => ram = value; }

        public Motherboard(MotherboardTypes motherboardTypes, RAMTypes rAMType, CPUTypes cPUTypes, int usingPover) : base(usingPover)
        {
            this.motherboardTypes = motherboardTypes;
            this.rAMType = rAMType;
            this.cPUTypes = cPUTypes;
            
        }

        public override string ToString()
        {
            String toReturn = "";
            if (cpu == null)
            {
                toReturn += "cpu" + "null";
            }
            else
            {
                toReturn += "cpu" + cpu.ToString();
            }
            if (ram == null)
            {
                toReturn += "ram" + "null";
            }
            else
            {
                toReturn += "ram" + ram.ToString();
            }
            return toReturn;
        }

        public bool isComponCorrect()
        {
            if (cpu==null||ram==null)
            {
                
                return false;
            }
            if (cpu.CpuTypes.Equals(cPUTypes)&&ram.RAMTypes.Equals(rAMType))
            {
                
                return true;
            }else{
                return false;
            }

        }
        public int countPowerOfMatherbordAndComponents()
        {
            return this.UsingPover + cpu.UsingPover + ram.UsingPover;
        }
    }
}

File: Lab1/Lab1/lab1j/src/model/Motherboard.java
package model;

import dto.CPUTypes;
import dto.MotherboardDto;
import dto.MotherboardTypes;
import dto.RAMTypes;

class Motherboard implements IAdapter<MotherboardDto>
    {
        private MotherboardDto motherboardDtoAdaptee;

        public MotherboardTypes MotherboardTypes (){
            return null;}
        public RAMTypes RAMType (){
        return null; }
        public dto.CPUTypes CPUTypes() {
        return null;}
        public CPU Cpu()
        {
            return null; }

        public RAM Ram() { return null;}

        public Motherboard(MotherboardDto motherboardDto)
        {
            motherboardDtoAdaptee = motherboardDto;
        }

        public Motherboard(MotherboardTypes motherboardTypes, RAMTypes rAMType, CPUTypes cPUTypes, int usingPover)
        {


        }

        public MotherboardDto MotherboardDtoAdaptee()
        {
            return null;
        }

        public String ToString()
        {
            String toReturn = "";

            return toReturn;
        }

        public boolean isComponCorrect()
        {
            if (motherboardDtoAdaptee.CpuDto==null||motherboardDtoAdaptee.RamDto==null)
            {

                return false;
            }
            if (motherboardDtoAdaptee.CpuDto.CpuTypes.Equals(motherboardDtoAdaptee.CPUTypes)
                &&motherboardDtoAdaptee.RamDto.RAMTypes.Equals(motherboardDtoAdaptee.RAMType))
            {

                return true;
            }else{
                return false;
            }
        }
        public int countPowerOfMatherbordAndComponents()
        {
            return motherboardDtoAdaptee.UsingPover + motherboardDtoAdaptee.CpuDto.UsingPover
                                                    + motherboardDtoAdaptee.RamDto.UsingPover;
        }

        public MotherboardDto GetAdaptee()
        {
            return motherboardDtoAdaptee;
        }

        public void SetAdaptee(MotherboardDto adaptee)
        {
            motherboardDtoAdaptee = adaptee;
        }

        public Motherboard()
        {
        }
    }

